### PR TITLE
docs: cdl-cli basic usage and schema-repo interaction

### LINKED ISSUE #88: docs: cdl-cli basic usage and schema-repo interaction
Add docs for cdl-cli

### PR DESCRIPTION
**Pull request by esavier**
fixes #88

### CODE REVIEW

File: docs/CDL-Cli.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+## How to guide:
+
+#### manipulate views
+views are a WIP feature, currently not used widely beside some cases in development.
+
+#### manipulate schemas
+
+###### add schema
+`cdl --registry-address "http:/localhost:6400 schema <add|get|names|update> --name <schemaname> \
```
Comment (Kononnable) [inline docs/CDL-Cli.md:20]:
```suggestion
`cdl --registry-address "http://localhost:6400 schema <add|get|names|update> --name <schemaname> \
```

File: docs/CDL-Cli.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Command Service
+
+## Technical Description
+
+Cdl-cli is currently the only tool able to interact with Schema-Registry. Its purpose is to interact with the Schema-Registry database by setting and viewing its content.
```
Comment (Kononnable) [inline docs/CDL-Cli.md:5]:
```suggestion
Cdl-cli is currently the only tool able to interact with schema registry. Its purpose is to interact with the schema registry database by setting and viewing its content.
```
I think we never use schema registry written as crate name in the docs.

File: docs/CDL-Cli.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+Please mind that this tool is a reference. Schema-Registry port should be configured open, and every tool that is able to communicate to it should be able to set and get information about schemas from Schema-Registry. Currently only schema-regisry only supports GRPC protocol, but there are some progress with both REST and webgui parts, which may be presented in the future as GRPC's alternative.
```
Comment (Kononnable) [inline docs/CDL-Cli.md:7]:
```suggestion
Please mind that this tool is a reference. Schema registry port should be configured open, and every tool that is able to communicate to it should be able to set and get information about schemas from schema registry. Currently only schema regisry only supports GRPC protocol, but there are some progress with both REST and webgui parts, which may be presented in the future as GRPC's alternative.
```

File: docs/CDL-Cli.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+Communication Methods:
+- GRPC
+
+## How to guide:
+
+#### manipulate views
+views are a WIP feature, currently not used widely beside some cases in development.
```
Comment (Kononnable) [inline docs/CDL-Cli.md:15]:
```suggestion
Views are a WIP feature, currently not used widely beside some cases in development.
```

File: docs/CDL-Cli.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+Cdl-cli is currently the only tool able to interact with Schema-Registry. Its purpose is to interact with the Schema-Registry database by setting and viewing its content.
+
+Please mind that this tool is a reference. Schema-Registry port should be configured open, and every tool that is able to communicate to it should be able to set and get information about schemas from Schema-Registry. Currently only schema-regisry only supports GRPC protocol, but there are some progress with both REST and webgui parts, which may be presented in the future as GRPC's alternative.
+
+Communication Methods:
+- GRPC
+
+## How to guide:
+
+#### manipulate views
```
Comment (Kononnable) [inline docs/CDL-Cli.md:14]:
```suggestion
#### Manipulate views
```

File: docs/CDL-Cli.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+views are a WIP feature, currently not used widely beside some cases in development.
+
+#### manipulate schemas
```
Comment (Kononnable) [inline docs/CDL-Cli.md:17]:
```suggestion
#### Manipulate Schemas
```
empty section?

File: docs/CDL-Cli.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+###### add schema
```
Comment (Kononnable) [inline docs/CDL-Cli.md:19]:
```suggestion
###### Add schema
```

File: docs/CDL-Cli.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+`cdl --registry-address "http:/localhost:6400 schema <add|get|names|update> --name <schemaname> \
+    --query-address <query-service-uri>" \
+    --topic <ingest-topic> \
+    --file <optional:schema-path>
+`
+
+- if `--file` is provided, specific file must have valid json inside.
```
Comment (Kononnable) [inline docs/CDL-Cli.md:26]:
```suggestion
- If `--file` is provided, specific file must have valid json inside.
```

File: docs/CDL-Cli.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+- if `--file` is missing, you will be asked to provide json inside cli,
```
Comment (Kononnable) [inline docs/CDL-Cli.md:27]:
```suggestion
- If `--file` is missing, you will be asked to provide json inside cli,
```

File: docs/CDL-Cli.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  to do so, insety valid json, followed by newline, and then press `ctrl+d` to finish editing
```
Comment (Kononnable) [inline docs/CDL-Cli.md:28]:
```suggestion
  To do so, insert valid json, followed by newline, and then press `ctrl+d` to finish editing
```

File: docs/CDL-Cli.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+  minimal valid json schema is `{}`
+- random schema id will be assigned to the request
+
+###### list schemas
```
Comment (Kononnable) [inline docs/CDL-Cli.md:32]:
```suggestion
###### List schemas
```

File: docs/CDL-Cli.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+- will print all existing schema names and IDs:
```
Comment (Kononnable) [inline docs/CDL-Cli.md:33]:
```suggestion
Will print all existing schema names and IDs:
```

File: docs/CDL-Cli.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+`cdl --registry-address "http:/localhost:6400 schema names`
```
Comment (Kononnable) [inline docs/CDL-Cli.md]:
```suggestion
`cdl --registry-address "http://localhost:6400 schema_names`
```
Can we pass multiple names here?

File: docs/schema_registry.md [hunk 14]
```diff
@@ -2,7 +2,7 @@
 
 ### Technical Description
 
-Schema Registy (also, `SR` for short) is responible for storing configuration about the data types handled by CDL. Schema Registry is a persistant graph database, that can be interfaced via grpc (other means of interaction are in progresss). Currently there is no GUI nor TUI. User interaction consist of internal tool [cdl-cli][cdl-cli]. There are also basic replication ability built-in.
+Schema Registy (also, `SR` for short) is responible for storing configuration about the data types handled by CDL. Schema Registry is a persistant graph database, that can be interfaced via grpc (other means of interaction are in progresss). Currently there is no GUI nor TUI. User interaction consist of internal tool [CDL-cli][CDL-cli]. There are also basic replication ability built-in.
```
Comment (Kononnable) [inline docs/schema_registry.md:5]:
```suggestion
Schema registry (also, `SR` for short) is responsible for storing configuration about the data types handled by CDL. It is a persistent graph database, that can be queried via grpc (other means of interaction are in progress). Currently there is no GUI nor TUI. User interaction consist of internal tool [CDL-cli][CDL-cli]. There is also basic replication ability built-in.
```

File: docs/schema_registry.md [hunk 15]
```diff
@@ -2,7 +2,7 @@
...
 
 Interacts with:
 - nothing on its own
```
Comment (Kononnable) [inline docs/schema_registry.md:8]: Technically it checks if passed kafka topic exists. Not sure if we should mention it here as it's not a cdl service.

File: docs/CDL-Cli.md [hunk 16]
```diff
@@ -0,0 +1,34 @@
+# Command Service
+
+## Technical Description
+
+Cdl-cli is currently the only tool able to interact with schema registry. Its purpose is to interact with the schema registry database by setting and viewing its content.
```
Comment (USSAMOH2) [inline docs/CDL-Cli.md:5]: Both CDL and CLI are acronyms, and so both should be capitalized.

Comment (USSAMOH2) [inline docs/CDL-Cli.md:5]:
```suggestion
The CDL-CLI is the official tool for interacting with the CDL's Schema Registry, used both for viewing and manipulating schemas and their respective data.
```

File: docs/CDL-Cli.md (hunk 13 above)
Comment (esavier) [reply docs/CDL-Cli.md]: nah, i dont think so, 
i also don't know what would happen if multiple jsons are provided inside file or in body

File: docs/CDL-Cli.md [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+
+Please mind that this tool is a reference. Schema registry port should be configured open, and every tool that is able to communicate to it should be able to set and get information about schemas from schema registry. Currently only schema regisry only supports GRPC protocol, but there are some progress with both REST and webgui parts, which may be presented in the future as GRPC's alternative.
```
Comment (USSAMOH2) [inline docs/CDL-Cli.md:7]:
```suggestion
For this tool to work, please make sure that the Schema Registry's gRPC server is listening on a public port. Currently, the Schema Registry only exposes a gRPC API, which is faster than a JSON API but less convenient to use. There is some progress with a JSON API for convenience, as well as a TUI (terminal user interface) and a website.
```

File: docs/CDL-Cli.md [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+
+Communication Methods:
+- GRPC
+
+## How to guide:
+
+#### Manipulate views
```
Comment (USSAMOH2) [inline docs/CDL-Cli.md:14]:
```suggestion
#### Manipulate Views
```

File: docs/CDL-Cli.md [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+    --query-address <query-service-uri>" \
+    --topic <ingest-topic> \
+    --file <optional:schema-path>
+`
+
+- if `--file` is provided, specific file must have valid json inside.
+- if `--file` is missing, you will be asked to provide json inside cli,
+  to do so, insety valid json, followed by newline, and then press `ctrl+d` to finish editing
+- minimal valid json schema is `{}`
+- random schema id will be assigned to the request
```
Comment (USSAMOH2) [inline docs/CDL-Cli.md:30]:
```suggestion
- If `--file` is missing, the CLI will expect JSON to be piped in over `stdin`.
- A schema containing `true` will accept any valid JSON data.
- New schemas are assigned a random UUID on creation, which will be printed after a successful insert.
```

File: docs/schema_registry.md [hunk 20]
```diff
@@ -2,7 +2,7 @@
 
 ### Technical Description
 
-Schema Registy (also, `SR` for short) is responible for storing configuration about the data types handled by CDL. Schema Registry is a persistant graph database, that can be interfaced via grpc (other means of interaction are in progresss). Currently there is no GUI nor TUI. User interaction consist of internal tool [cdl-cli][cdl-cli]. There are also basic replication ability built-in.
+Schema registry (also, `SR` for short) is responsible for storing configuration about the data types handled by CDL. It is a persistent graph database, that can be queried via grpc (other means of interaction are in progress). Currently there is no GUI nor TUI. User interaction consist of internal tool [CDL-cli][CDL-cli]. There is also basic replication ability built-in.
```
Comment (USSAMOH2) [inline docs/schema_registry.md:5]:
```suggestion
The Schema Registry (`SR` for short) is responsible for storing configuration about the data types handled by CDL. It is a persistent graph database, that can be queried via gRPC (other means of interaction are in progress). Currently there is no GUI nor TUI; user interaction is currently performed with the [CDL-CLI][CDL-CLI]. Replication across multiple instances of the Schema Registry is supported.
```

File: docs/schema_registry.md [hunk 21]
```diff
@@ -38,4 +38,4 @@ Mind that GRPC uses HTTP2 as its transport protocol (L4), so SCHEMA_REGISTRY_ADD
 See an example [configuration][configuration] of deployment of data router and other services.
 
 [configuration]: ../examples/deploy/SETUP.md
-[cdl-cli]: cdl-cli.md
\ No newline at end of file
+[CDL-cli]: CDL-cli.md
```
Comment (USSAMOH2) [inline docs/schema_registry.md:41]:
```suggestion
[CDL-CLI]: CDL-CLI.md
```

File: docs/CDL-Cli.md [hunk 22]
```diff
@@ -0,0 +1,34 @@
...
+
+###### List schemas
+Will print all existing schema names and IDs:
+`cdl --registry-address "http:/localhost:6400 schema names`
```
Comment (USSAMOH2) [inline docs/CDL-Cli.md:34]:
```suggestion
###### List Schemas

 To print all existing schema names and their respective ID's:
`cdl --registry-address "http://localhost:6400 schema names`
```

**Review by USSAMOH2 — CHANGES_REQUESTED** [review]
You should also rename this document to `CDL-CLI.md`.

File: docs/schema_registry.md (hunk 15 above)
Comment (esavier) [reply docs/schema_registry.md:8]: i am leaving it out then?

**Review by luke-biel — APPROVED** [review]

**Review by Kononnable — DISMISSED** [review]

**Review by luke-biel — APPROVED** [review]

**Review by luke-biel — APPROVED** [review]

**Pull request merged**